src/coreclr/vm/peimagelayout.cpp

```diff
@@ -775,10 +775,15 @@ void* FlatImageLayout::LoadImageByCopyingParts(SIZE_T* m_imageParts) const
     }
 #endif // FEATURE_ENABLE_NO_ADDRESS_SPACE_RANDOMIZATION
 
+    DWORD allocationType = MEM_RESERVE | MEM_COMMIT;
+#if defined(__APPLE__) && defined(HOST_ARM64)
+    allocationType |= MEM_RESERVE_EXECUTABLE;
```

Comment:
I would do this for `HOST_UNIX` in general like we do in the `ExecutableAllocator::Reserve`.

Reply:
That would cause that memory to be taken from the pre-reserved block in PAL that is close to the libcoreclr.so.

Reply:
Will it help R2R code in the same way as JITed code (making it closer to coreclr and avoiding stubs) ?

Reply:
Did not see you've answered this already.